frontend/src/components/dashboard/create/sms/SMSValidationInput.tsx

```diff
@@ -16,8 +16,8 @@ const EmailValidationInput = ({
   }
 
   async function onClickHandler() {
-    await onClick(recipient)
     setRecipient('')
```

Comment:
Instead of switching the order, I think it's possible to just get rid of the `setRecipient('')` call instead. Since if validation succeeds, it will be unmounted when the next step is shown. Even if the user chooses to enter new credentials again, the component will be reinitialised with an empty value, negating a need to manually reset it's value.

This way, we can still preserve the value in the input when sending. Otherwise if there's an error, the user might not be able to see what he/she had input in the first place. 

![image](https://user-images.githubusercontent.com/3666479/100591186-5e642400-3330-11eb-8469-67e034594e35.png)

Reply:
I agree, tried it out and it works as intended